pallets/gear/src/manager/journal.rs

```diff
@@ -160,8 +155,13 @@ where
 
                 self.clean_reservation_tasks(program_id, maybe_inactive);
 
-                ProgramStorageOf::<T>::update_program_if_active(program_id, |p, bn_ref| {
-                    *bn_ref = Pallet::<T>::block_number();
+                ProgramStorageOf::<T>::update_program_if_active(program_id, |p, bn| {
+                    let _ = TaskPoolOf::<T>::delete(
+                        *bn,
+                        ScheduledTask::PauseProgram(program_id),
+                    );
+
+                    *bn = Pallet::<T>::block_number();
```

Comment:
add todo to move expiration to ActiveProgram

Reply:
https://github.com/gear-tech/gear/issues/2570